liferay-headless/liferay-portal
# LPS-168462: Add nested entities to OpenAPI schemas

**What is new?**
- Add `objectRelationshipName` as a property in all 1-M relationships to show the nested entity.
- Display at level 1 all nested entities in OpenAPI schemas.
- These changes support nested entities for Custom and System Object Definitions. 

**NOTE:** All schemas will contain the `objectRelationshipName` as nested entities for M-M and 1-M relationship

**How to reproduce this?**
1. Deploy `object-rest-impl`
2. Enable the following feature flag: `feature.flag.LPS-168462`
3. Create two Custom Objects, like Student and Subject, and relate them with a M-M relationship, `studentSubject`.
4. Go to `/o/api/subjects` and see Subject's schema:

![image](https://user-images.githubusercontent.com/44723449/212719054-4c48cf55-0d39-4093-a4b5-ff93eccaf90f.png)

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-168462
